File: solutions/leetcode_451.swift
class Solution {
    func frequencySort(_ s: String) -> String {
        var dict = [Character:Int]()
        for c in s {
            dict[c] = dict[c, default:0] + 1
        }

        var list = [(Character,Int)]()
        for (c,count) in dict {
            list.append((c,count))
        }

        list.sort{a,b in
            a.1 > b.1
        }

        var result = ""
        for (c,count) in list {
            result += String(repeating: c, count: count)
        }

        return result
    }
}

File: solutions/leetcode_451.py
class Solution:
    def frequencySort(self, s: str) -> str:
        chars = {}
        for c in s:
            chars[c] = chars.get(c, 0) + 1
        chars = sorted(chars.items(), key=lambda k:k[1], reverse=True)
        return ''.join(c*count for c,count in chars)


if __name__ == '__main__':
    given_n = "tree"
    expected = "eetr"

    solution = Solution()
    result = solution.frequencySort(given_n)
    assert result == expected
